### PR TITLE
Add support for more m.room.events in timeline

### PR DESCRIPTION
**Pull request by adasauce**
- add delegate choice for more events:
    - room create
    - power levels
    - room join rules
    - room history visibility
    - room guest access 
 - rename some RoomEventType structs to be more consistent
 - added reason: formatting to membership events arbitrarily
 - added translations

### CODE REVIEW

**Comment by adasauce** [comment]
been using nheko with my PR and all is lookin good. haven't found anything broken with the changes. there's still a few more room events to add support for but I think I'll cut this here and move onto the event silencing stuff in a separate branch, that's what I was working toward anyway, and this is nice and isolated from that code.

File: src/timeline/TimelineModel.cpp [hunk 1]
```diff
@@ -1471,6 +1474,107 @@ TimelineModel::formatTypingUsers(const std::vector<QString> &users, QColor bg)
...
+        auto event =
+          std::get_if<mtx::events::StateEvent<mtx::events::state::PowerLevels>>(&events[id]);
+        if (!event)
+                return "";
+
+        QString user = QString::fromStdString(event->sender);
+        QString name = escapeEmoji(displayName(user));
+
+        // TODO: power levels rendering is actually a bit complex. work on this later.
+        return tr("%1 has changed the room's permissions.").arg(name);
```
Comment (MayeulC) [inline src/timeline/TimelineModel.cpp]: If you wanted to make the changes from https://github.com/Nheko-Reborn/nheko/pull/131/files#diff-635ee45863a598079569b6d91d50c8e1R1574-R1582 work with the translation, you need something like:

```suggestion
        QString* message[] = {
                              //: Comment for the translator: one of the following strings is randomly chosen on powerlevel changes, have fun!
                               tr("%1 has altered the deal, pray they do not alter it any further."),
                               tr("%1 has increased power levels to over 9000!"),
                               tr("%1 rolls for a power check, critical hit."),
                               tr("%1 has tipped the scales."),
                               tr("%1 thinks the rules need to change.")
                               };

        QString *chosen = message[rand() % sizeof(message)];

        // TODO: power levels rendering is actually a bit complex. work on this later.
        return chosen->arg(name);
```

(this is untested). I am not sure how to allow a different number of translated strings in multiple languages.

Comment (adasauce) [reply src/timeline/TimelineModel.cpp]: thanks! hope thisdoesn't get lost if we decide to re-inject jokes before the power levels stuff gets fully implemented.

**Comment by adasauce** [comment]
don't look at me like that, I merged my own PR. :fearful:

**Pull request merged**